### ISSUE TITLE
GPU Particle System

### ISSUE THREAD

**Issue by vilbeyli** [vilbeyli/DX11Renderer#42]

**Comment by AraHaan** [comment]
Does this support ``*.fx`` files to?

**Comment by vilbeyli** [comment]
No, for the time being, shaders are separate *.hlsl files only. I wanted to keep the number of third-party/tool libraries as low as possible.

**Comment by AraHaan** [comment]
ok, good to know because I was thinking about using this with the x2.fx file from Elsword on a test project to see what it does. Unless there is a way to convert it over. 👀

**Comment by vilbeyli** [comment]
FX API is HLSL-based, it shouldn't be hard to convert the shaders. I'm not very familiar with the FX system, but I remember there are passes. This will probably require runtime logic, which means one has to edit Engine/Renderer/RenderPasses.cpp/h to pass the data to shaders and create passes.

I'll add a short demonstration with minimal code about how to use the renderer to Readme soon, which hopefully would help with such changes.
